prov/sm2/src/sm2_atomic.c

```diff
@@ -109,9 +109,9 @@ sm2_do_atomic_inject(struct sm2_ep *ep, int64_t peer_gid, uint32_t op,
 		return ret;
 
 	xfer_entry->hdr.proto = sm2_proto_inject;
-	xfer_entry->hdr.proto_flags |= proto_flags;
 
 	sm2_generic_format(xfer_entry, ep->gid, op, 0, 0, op_flags, context);
+	xfer_entry->hdr.proto_flags |= proto_flags;
```

Comment:
seems like this should just be fixed in the first patch

Reply:
The first patch introduced a new proto flag but this patch fixed a bug in how atomics are using proto flags. I think they are different enough to be in separate commits. I can squash them if you want.